lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -176,13 +176,33 @@ def credit(amount, payment, options = {})
         end
       end
 
-      def verify(credit_card, options = {})
+      def verify(payment_method, options = {})
+
+        amount = define_amount_for_verify(payment_method, options)
+
         MultiResponse.run(:use_first_response) do |r|
-          r.process { authorize(100, credit_card, options) }
-          r.process(:ignore_result) { void(r.authorization, options) }
+          r.process { authorize(amount, payment_method, options) }
+          r.process(:ignore_result) { void(r.authorization, options) } unless amount == 0
         end
       end
 
+      def define_amount_for_verify(payment_method, options)
```

Comment:
💬 
maybe you can refactor a little bit this function.
```suggestion
      def define_amount_for_verify(payment_method, options)
        return 100 if payment_method.is_a?(CreditCard) && payment_method.brand == 'jcb'
     
        if options[:verify_amount].present? && payment_method.is_a?(CreditCard)
          requires!(options, :billing_address)
          requires!(options[:billing_address], :address1, :zip)
          return options[:verify_amount]
        end
        raise ArgumentError, 'verify_amount value must be an integer and greater than zero' if verify_amount_value_verification(options)        
      end
```

Reply:
code has been refactored, so please check again